solution jv-array-list

ASelivonchyk opened the pull request:
- implemented custom ArrayList.class according to the task;

ASelivonchyk committed:
solution jv-array-list
- implemented custom ArrayList.class according to the task;

andriy-shulepa commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,99 @@
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
```
The error message in the ArrayListIndexOutOfBoundsException should include the invalid index value.

andriy-shulepa commented on src/main/java/core/basesyntax/ArrayList.java:
```diff
@@ -1,48 +1,99 @@
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(elementsData[i], element)) {
```
fix common mistake
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-use-class-objects

andriy-shulepa reviewed (changes requested):
next time, create new branch and push your code there instead of pushing directly to master branch

ASelivonchyk committed:
fixed codestyle.
- added invalid index value to error message in the ArrayListIndexOutOfBoundsException in checkIndex() method;
- deleted usage of Objects.class;

ASelivonchyk committed:
fixed codestyle.
